### ISSUE TITLE
Pod file

### ISSUE THREAD

**Issue by dimohamdy** [graphhopper/graphhopper-ios#7]
Add graphhopper to CocoaPods

**Comment by ntpeters** [comment]
+1

**Comment by clns** [comment]
Adding graphhopper-ios to CocoaPods means translating the java sources into Objective-C (which this repo does) and using the translated files to create the library. I absolutely agree with this, however we first need to have the java tests running on iOS ( #1 ) to make sure the library is fully functional.

At this point there is only a [small demo](../tree/master/graphhopper-ios-sample) that uses basic functionality from GraphHopper. We need the tests to make sure everything else is working. It's a bit more difficult to translate the tests because of the [stuff that had to be commented out](/graphhopper/graphhopper/compare/0.4...ios-compat) to make it work on iOS.

I'll try to start investigating how can we translate the tests. I'm open to any suggestions/contributions and I'd certainly appreciate them.

**Comment by karussell** [comment]
The XML stuff is not that important on iOS, we do not have the import also not on Android. So if the core tests like the storage and the algorithms work this would be already a nice start. Is it possible to run device-independent Objective-C code on linux? This way I could easier help and we could use travis ci for it, so that it automatically runs after a push :)

**Comment by clns** [comment]
Yes, once we have them we can definitely use Travis. Not sure about running objc code on linux, it might be possible with GNUstep but I don't have any experience with that.

Sounds like it might not be that difficult. Can you outline the tests that are most important to have on iOS? So I can try to translate them and see where we are after that :smile:

**Comment by karussell** [comment]
It looks to me like all tests, except the [reader package](https://github.com/graphhopper/graphhopper/tree/master/core/src/test/java/com/graphhopper/reader)

Or expressed the other way around: at least DijkstraBidirectionRefTest.java and all [storage](https://github.com/graphhopper/graphhopper/tree/master/core/src/test/java/com/graphhopper/storage) tests would be needed (without unsafe)

Also PrepareContractionHierarchies (and its tests) is not necessary for iOS BUT it would be still nice to have the class (and its tests) so that we can run the DijkstraBidirectionCHTest.java

Also here I would avoid as much manual intervention as possible :)
E.g. I plan a bigger refactoring where the LevelGraphStorageTest won't exist

**Comment by clns** [comment]
Avoiding manual intervention sounds great to me :smile: Also I think we can get rid of most of the current "manual intervention" when the separation between the import and core is made.

**Comment by karussell** [comment]
Ok, created an issue for this https://github.com/graphhopper/graphhopper/issues/450

**Comment by karussell** [comment]
Why close this?

**Comment by dimohamdy** [comment]
i think it's not in the scope now

**Comment by karussell** [comment]
In which scope? We should and will make this happen, sometimes it takes a bit more time :)

**Comment by dimohamdy** [comment]
:+1:

**Comment by clns** [comment]
As @karussell said, we are going to do this. But to be able to publish it as a library on [CocoaPods](https://cocoapods.org/) it needs to be stable and reliable, meaning we need to have the tests pass ( #1 ).

The major problem at the moment is that there's some [code we had to strip out from the iOS port (for GraphHopper 0.5)](https://github.com/graphhopper/graphhopper/compare/0.5...ios-compat) to make it work. This is related to splitting up the import stuff from the core (https://github.com/graphhopper/graphhopper/issues/450), which will happen in a future release of GraphHopper. Once we have this, we will be able to refactor the iOS version and move forward with the tests.

**Comment by karussell** [comment]
BTW: CocoaPods might not be always desired https://tech.zalando.com/blog/how-the-zalando-ios-app-abandoned-cocoapods-and-reduced-build-time/

**Comment by RandyHBH** [comment]
how is it going the cocoapods spec?

**Comment by RaimundWege** [comment]
One year later: how is it going? Currently we're building an App with Flutter and I'm trying to build a dart package for graphhopper but first I need a pod source. For now I will use a git repository as pod dependency but an official source would be nice.
